Bold links are fixed by removing special symbols

Kostov6 opened the pull request:
**What this PR does / why we need it**:
This PR fixes the issue with bolding autolinks. It is done by removing special symbols from the postfix of the autolink based on this [documentation](https://github.github.com/gfm/#extended-autolink-path-validation)

**Which issue(s) this PR fixes**:
Fixes #217

**Special notes for your reviewer**:
@Kristian-ZH 

**Release note**:

<!--  Write your release note:
1. Enter your release note in the below block.
2. If no release note is required, just write "NONE" within the block.

Format of block header: <category> <target_group>
Possible values:
- category:       improvement|noteworthy|action
- target_group:   user|operator|developer
-->
```noteworthy user
None
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/gardener/docforge?pullRequest=220) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/gardener/docforge?pullRequest=220) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/gardener/docforge?pullRequest=220) it.</sub>

Kristian-ZH reviewed (commented):
Thanks for the PR,

In my opinion, the punctuation symbol (`?` and `.` )can be added [here](https://github.com/gardener/docforge/blob/d6987977b6bcce88e3e09b994ce29cafec99f17c/pkg/markdown/parser/autolinks_parse.go#L241)

Also in the Markdown, I do not know an example where `:` and `~` are used. (Refer to https://www.markdownguide.org/basic-syntax/)
Also based on the link above, In the markdown we can have the following syntax \`Some text\` so ` should be also escaped.

So... In my opinion, the added code should look in the following way:
```
	for ;  data[linkEnd-1] == '*' || data[linkEnd-1] == '_' || data[linkEnd-1] == '`'; linkEnd-- {
	}
```
and the punctuation symbols should be added in the punctuation section.


/cc @dimitar-kostadinov

Kostov6 commented:
Thank you for the review,

I've made the changes you have recommended in the review:
1. added ? and ! in the punctuation check
2. changed the for loop as recommended

Kristian-ZH reviewed (approved):
/lgtm
